src/sonic-yang-models/yang-models/sonic-bgp-device-global.yang

```diff
@@ -21,12 +21,26 @@ module sonic-bgp-device-global {
         container BGP_DEVICE_GLOBAL {
             container STATE {
                 description "BGP device-specific global data";
-                leaf tsa_enabled {
+                leaf tsa_enabled {                    
                     type boolean;
                     default "false";
                     description
                     "When set to true, Traffic is shifted away (TSA), i.e, BGP routes are not advertised to neighboring routers";
                 }
+                leaf idf_isolation_state {                    
```

Comment:
>idf_isolation_state [](http://example.com/codeflow?start=21&length=19)

Is there any limitation of this field on M0/T0/T1 devices? If no, could we generalize the isolation for all roles, not to mention IDF?

Reply:
IDF isolation is only supported on T2 and the config will not have any effect on T0/T1 devices.

Reply:
Then can we enforce a value for M0/T0/T1? The yang model could help the enforcement.

Reply:
Currently, the field is initialized to "unisolated" on all devices (including M0/T0/T1) in init_cfg.json.j2.  Changing this configuration through CLI/Script is blocked on non-T2 devices. Is that sufficient?

Reply:
What is the backend behavior if a T0 got config which is not unisolated? will it fail or ignore?

If the backend will treat this as a failure, we can also hardening yang models, so when ConfigDB is validated by yang models, the failure could be detected earlier, and stop the configuration process immediately.

Reply:
Only T2 will handle this config change. T0/T1 will ignore it